### DISCUSSION TITLE
When to use AddForce, and when to use AddImpulse

### DISCUSSION THREAD

**Discussion by joshcoyne** [jrouwe/JoltPhysics#1196] [General]
hello!

I'm doing some tests where I want to simulate a flying vehicle. I've had the most predictable success using AddForce to get enough upward momentum to beat gravity. That said, I'm not sure what the situations are for Force and Impulse and would like some details :)

**Comment by DocAce** [comment]
Force is scaled to the time step, so if you have something acting over time you should use it for consistent results. Impulse is not scaled and useful for one-time pushes.

**Comment by jrouwe** [comment]
Also, AddForce doesn't change the velocity of the body until the next PhysicsSystem::Update, AddImpulse immediately changes velocity. Other than that you can pretty much achieve the same results with both functions: Impulse = Force * Delta Time
